dbgd.c

```diff
@@ -211,21 +211,55 @@ static int dbgd_mem_read(Dbg__Request *req, Dbg__Response *res)
     res->address = req->address;
     res->has_address = 1;
 
-    res->size = 1; /* FIXME: add word, dword, qword support */
-    res->has_size = 1;
+    res->data.len  = req->size;
+    res->data.data = malloc(res->data.len);
```

Comment:
That's what I thought (for the sending functions at least). I guess an option is to pass some kind of allocator function to it - anything allocated through that function would later be free'd.
The alternative is to hardcode cleanups for all `bytes` type variables.


**Are you fine with me not fixing, but documenting this leak in an issue after merge, to avoid a delay for this feature?**

Backstory: I'll add a CALL method in a follow-up PR; once all of that works, my AC97 unit test finally works too. Then I can *finally* PR my xqemu code I'm holding back until the unit test works on nxdk/nxdk-rdt master.
So, hopefully understandably, I don't want to make experimental changes just to fix a "small" memory leak while other changes pile up